File: java/src/com/vo/Zone.java
package com.vo;

import java.util.Set;

public class Zone
{
    private String name;
    private Set<Counter> counters;
    private Integer totalIn = 0;
    private Integer totalOut = 0;
    private Integer occupancy = 0;

    public Zone(String name, Set<Counter> counters) {
        this.name = name;
        this.counters = counters;
    }

    public void update() {
        totalIn = 0;
        totalOut = 0;
        for (Counter c : counters) {
            c.update();
            totalIn += c.getTotalIn();
            totalOut += c.getTotalOut();
        }
        occupancy = totalIn - totalOut;
    }



}

File: kotlin/src/main/kotlin/Zone.kt
package com.vo

class Zone(private val name: String, private val cameras: List<Camera>) {
    private var currentCount = CurrentCount(0, 0)
    val occupancy get() = currentCount.total

    fun update() {
        currentCount = cameras.fold(CurrentCount(0, 0)) {currentCount, camera ->
            currentCount.add(camera.currentCount)
        }
    }
}